hub/saml-engine/src/integration-test/java/uk/gov/ida/integrationtest/hub/samlengine/apprule/RpAuthnResponseGeneratorResourceTest.java

```diff
@@ -185,4 +187,102 @@ public void shouldReturnAnErrorResponseGivenBadInput() throws JsonProcessingExce
         ErrorStatusDto errorStatusDto = rpAuthnResponse.readEntity(ErrorStatusDto.class);
         assertThat(errorStatusDto.getExceptionType()).isEqualTo(ExceptionType.INVALID_INPUT);
     }
+
+    @Test
+    public void shouldUseEncryptedAssertionsFromListIfNonEmpty() throws Exception {
+        // Given
+        ResponseFromHubDto responseFromHubDto = aResponseFromHubDto()
+            .withAuthnRequestIssuerEntityId(TestEntityIds.TEST_RP)
+            .withAssertions(Arrays.asList(createAssertionString(), createAssertionString()))
+            .build();
+        configStub.setupCertificatesForEntity(responseFromHubDto.getAuthnRequestIssuerEntityId());
+        configStub.signResponsesAndUseSamlStandard(responseFromHubDto.getAuthnRequestIssuerEntityId());
+
+        // When
+        URI generateAuthnResponseEndpoint = samlEngineAppRule.getUri(Urls.SamlEngineUrls.GENERATE_RP_AUTHN_RESPONSE_RESOURCE);
+        Response rpAuthnResponse = client.target(generateAuthnResponseEndpoint).request().post(Entity.entity(responseFromHubDto, MediaType.APPLICATION_JSON_TYPE));
+
+        // Then
+        assertThat(rpAuthnResponse.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
+        AuthnResponseFromHubContainerDto result = rpAuthnResponse.readEntity(AuthnResponseFromHubContainerDto
+            .class);
+
+        org.opensaml.saml.saml2.core.Response response = extractResponse(result);
+        assertThat(response.getStatus().getStatusCode().getValue()).isEqualTo(StatusCode.SUCCESS);
+        assertThat(response.getEncryptedAssertions().size()).isEqualTo(2);
```

Comment:
I would like to do better than just checking the size, but I'm not sure how to mock the encryption etc.